libs/core/langchain_core/messages/utils.py

```diff
@@ -255,7 +258,11 @@ def _convert_to_message(message: MessageLikeRepresentation) -> BaseMessage:
 
 
 def convert_to_messages(
-    messages: Sequence[MessageLikeRepresentation],
+    messages: Union[
+        Iterable[MessageLikeRepresentation],
+        PromptValue,
+        BaseChatPromptTemplate,
```

Comment:
whats the use case for converting an unformatted prompt?

Reply:
In my specific instance, I have some shared prompting which is occasionally augmented. This results in the occasional sequences as follows:

```
SystemMessage
SystemMessage
HumanMessage
AIMessagePromptTemplate
(maybe another AIMessagePromptTemplate)
HumanMessagePromptTemplate
...
```

Reply:
not sure i fully understand, even in this case couldn't you do the merging after the prompt template has been converted to a prompt value?

Reply:
if its about using `+`, this is intended to be used for prompt creation, not for combining completed messages. those should just be appended to a list